extensions/positron-assistant/src/test/anthropic.test.ts

```diff
@@ -0,0 +1,146 @@
+		const emptyTextPart = new vscode.LanguageModelTextPart('');
+		const nonEmptyTextPart = new vscode.LanguageModelTextPart('Hello');
+
+		const messagesWithEmpty = [
+			// Message with only empty text content
+			new vscode.LanguageModelChatMessage(vscode.LanguageModelChatMessageRole.User, [emptyTextPart]),
+			// Message with non-empty text content
+			new vscode.LanguageModelChatMessage(vscode.LanguageModelChatMessageRole.User, [nonEmptyTextPart]),
+			// Message with both empty and non-empty text content
+			new vscode.LanguageModelChatMessage(vscode.LanguageModelChatMessageRole.Assistant, [emptyTextPart, nonEmptyTextPart])
+		];
+
+		// Call the method under test
+		await model.provideLanguageModelResponse(
+			messagesWithEmpty,
+			{},
+			'test-extension',
+			mockProgress,
+			mockCancellationToken
+		);
+
+		// Check that messages were filtered before being passed to the Anthropic client
+		const streamCall = mockClient.messages.stream.getCall(0);
+		assert.ok(streamCall, 'Stream method was not called');
+
+		// The empty message should be filtered out
+		const messagesPassedToAnthropicClient = streamCall.args[0].messages;
+		assert.strictEqual(messagesPassedToAnthropicClient.length, 2, 'Only non-empty messages should be passed to the Anthropic client');
+
+		// Verify that the message with only empty content was filtered out
+		const hasEmptyMessage = messagesPassedToAnthropicClient.some((msg: any) => {
+			return msg.content.length === 0 ||
+				(msg.content.length === 1 &&
+					msg.content[0].type === 'text' &&
+					msg.content[0].text === '');
+		});
+		assert.strictEqual(hasEmptyMessage, false, 'Messages with only empty content should be filtered out');
+	});
+
+	/**
+	 * Test the filtering behavior by checking the messages passed to Anthropic
+	 * when a message with empty content is included.
+	 */
+	test('provideLanguageModelResponse filters empty messages correctly with different content mixes', async () => {
+		// Create test messages with different combinations of empty/non-empty content
+		const emptyTextPart = new vscode.LanguageModelTextPart('');
+		const whitespaceTextPart = new vscode.LanguageModelTextPart('   \n  ');
+		const nonEmptyTextPart = new vscode.LanguageModelTextPart('Hello');
+
+		const messagesWithVariousContent = [
+			// Message with only empty text content - should be filtered out
+			new vscode.LanguageModelChatMessage(vscode.LanguageModelChatMessageRole.User, [emptyTextPart]),
+			// Message with only whitespace - should be filtered out
+			new vscode.LanguageModelChatMessage(vscode.LanguageModelChatMessageRole.User, [whitespaceTextPart]),
+			// Message with non-empty text content - should be kept
+			new vscode.LanguageModelChatMessage(vscode.LanguageModelChatMessageRole.User, [nonEmptyTextPart]),
+			// Message with both empty and non-empty text content - should be kept
+			new vscode.LanguageModelChatMessage(vscode.LanguageModelChatMessageRole.Assistant, [emptyTextPart, nonEmptyTextPart])
```

Comment:
Nit: can also use the short-hand
```suggestion
			// Message with only empty text content - should be filtered out
			vscode.LanguageModelChatMessage.User([emptyTextPart]),
			// Message with only whitespace - should be filtered out
			vscode.LanguageModelChatMessage.User([whitespaceTextPart]),
			// Message with non-empty text content - should be kept
			vscode.LanguageModelChatMessage.User([nonEmptyTextPart])
			// Message with both empty and non-empty text content - should be kept
			vscode.LanguageModelChatMessage.Assistant([emptyTextPart, nonEmptyTextPart])
```